src/Storage/Internal/StorageDataClient.php

```diff
@@ -0,0 +1,263 @@
+     */
+    private function _put(string $storeName, string $key, $value, string $type): ResponseFuture
+    {
+        try {
+            validateStoreName($storeName);
+            validateNullOrEmpty($key, "Key");
+            $storeValue = new _StoreValue();
+            if ($type == StorageValueType::INTEGER) {
+                $storeValue->setIntegerValue($value);
+            } elseif ($type == StorageValueType::DOUBLE) {
+                $storeValue->setDoubleValue($value);
+            } elseif ($type == StorageValueType::STRING) {
+                $storeValue->setStringValue($value);
+            } elseif ($type == StorageValueType::BYTES) {
+                $storeValue->setBytesValue($value);
+            }
+            $putRequest = new _StorePutRequest();
+            $putRequest->setKey($key);
+            $putRequest->setValue($storeValue);
+            $call = $this->grpcManager->client->Put(
+                $putRequest,
+                ["store" => [$storeName]],
+                ['timeout' => $this->timeout]
+            );
+        } catch (SdkError $e) {
+            return ResponseFuture::createResolved(new StoragePutError($e));
+        } catch (Exception $e) {
+            return ResponseFuture::createResolved(new StoragePutError(new UnknownError($e->getMessage(), 0, $e)));
+        }
+
+        return ResponseFuture::createPending(
+            function () use ($call): StoragePutResponse {
+                try {
+                    $this->processCall($call);
+                } catch (SdkError $e) {
+                    return new StoragePutError($e);
+                } catch (Exception $e) {
+                    return new StoragePutError(new UnknownError($e->getMessage(), 0, $e));
+                }
+                return new StoragePutSuccess();
+            }
+        );
+    }
+
+    /**
+     * @param string $storeName
+     * @param string $key
+     * @return ResponseFuture<StorageGetResponse>
+     */
+    public function get(string $storeName, string $key): ResponseFuture
+    {
+        try {
+            validateStoreName($storeName);
+            validateNullOrEmpty($key, "Key");
+            $getRequest = new _StoreGetRequest();
+            $getRequest->setKey($key);
+            $call = $this->grpcManager->client->Get(
+                $getRequest,
+                ["store" => [$storeName]],
+                ['timeout' => $this->timeout]
+            );
+        } catch (SdkError $e) {
+            return ResponseFuture::createResolved(new StorageGetError($e));
+        } catch (Exception $e) {
+            return ResponseFuture::createResolved(new StorageGetError(new UnknownError($e->getMessage(), 0, $e)));
+        }
+
+        return ResponseFuture::createPending(
+            function () use ($call): StorageGetResponse {
+                try {
+                    $response = $this->processCall($call);
+                } catch (SdkError $e) {
+                    return new StorageGetError($e);
```

Comment:
I'll be adding logic here to differentiate between `Element not found` and the rest of the errors. I'll also modify the `StorageGetResponse` and `StorageGetSuccess` objects to add a way to signify that the element was not found.